tartufo/scanner.py

```diff
@@ -142,11 +142,46 @@ class ScannerBase(abc.ABC):  # pylint: disable=too-many-instance-attributes
     global_options: types.GlobalOptions
     logger: logging.Logger
     _scan_lock: threading.Lock = threading.Lock()
+    _hex_entropy_score: float = 3.0
+    _b64_entropy_score: float = 4.5
 
     def __init__(self, options: types.GlobalOptions) -> None:
         self.global_options = options
         self.logger = logging.getLogger(__name__)
 
+        # entropy_score is a probability (between 0.0 and 1.0) that a given string
+        # is random. Strings that are at least this likely to be random will result
+        # in findings. We convert this from "sensitivity" (0-100) which is inverted
+        # so that intuitively "more sensitive" means "more likely to flag a given
+        # string as suspicious."
+        if self.global_options.entropy_sensitivity is None:
+            sensitivity = 25
+        else:
+            sensitivity = self.global_options.entropy_sensitivity
+        entropy_score = float(100 - sensitivity) / 100.0
+
+        # We now compute an effective score for each type of entropy string by
+        # multiplying by the number of bits expressed in each character of the
+        # string's character set:
+        # hex character 0-f is 16 digits = 2^4 digits = 4 bits/character
+        # base64 represents 24 bits using 4 characters = 6 bits/character
+        self._hex_entropy_score = entropy_score * 4.0
+        self._b64_entropy_score = entropy_score * 6.0
+
+        # For backwards compatibility, allow the caller to manipulate each of
+        # these representation-specific scores directly (but complain about it).
+        if self.global_options.hex_entropy_score:
+            warnings.warn(
+                "--hex-entropy-score is deprecated. Use --entropy-sensitivity instead."
+            )
+            self._hex_entropy_score = self.global_options.hex_entropy_score
+
+        if self.global_options.b64_entropy_score:
+            warnings.warn(
+                "--b64-entropy-score is deprecated. Use --entropy-sensitivity instead."
+            )
```

Comment:
```suggestion
        if self.global_options.hex_entropy_score:
            warnings.warn(
                "--hex-entropy-score is deprecated. Use --entropy-sensitivity instead.",
                DeprecationWarning
            )
            self._hex_entropy_score = self.global_options.hex_entropy_score

        if self.global_options.b64_entropy_score:
            warnings.warn(
                "--b64-entropy-score is deprecated. Use --entropy-sensitivity instead.",
                DeprecationWarning
            )
```
It's important that these are specifically deprecation warnings, or the user will likely not see them: https://github.com/godaddy/tartufo/blob/3b4acef809158676ebf7dfa8b93ae6f0a525b199/tartufo/cli.py#L277-L278

Reply:
I think it's fine for these deprecation warnings to stay here; I think the new calculations should move to properties though, staying consistent with all the rest of our calculated instance properties. Or of course the warnings can move with them. Either way. But I think let's keep our calculated things in properties :)

Reply:
Assuming you meant "expose these as properties instead of internal members", it's done now.

Reply:
And the warnings are fixed -- thanks for the information, I was clueless regarding the distinction or handling!

Reply:
I actually meant having the calculations in the properties as well, keeping the init as minimal as possible, and all calculations as JIT 😄 Same as we do with all our exclusion patterns, etc.

Reply:
The other advantage of using properties is that it cuts down on the number of instance variables, potentially appeasing pylint's warning

Reply:
I just added another commit which removes those two members at the cost of nearly doubling the related code. But pylint and mypy are pytest are happy, so it must be good. :)

Reply:
...and fixed again per this morning's review meeting.